File: Chapter3/tank_model.m
%
%   Model of the tank
%   Copyright Hamid D. Taghirad 2013
%
function xp=tank_model(t,x)
%   State variable x=l: Tank level
A=1; C=2.0;
F_in=0; % No disturbance input
u=0.1;  % Constant opening for valve 

xp=1/A*(F_in-C*u*sqrt(x));
end


 

File: Chapter3/tank_model.py
# Import necessary libraries
import numpy as np

# Define the model of the tank
def tank_model(x, t):
    A = 1.0
    C = 2.0
    F_in = 0.0  # No disturbance input
    u = 0.1     # Constant opening for valve
    
    x = np.maximum(x, 0)  # Ensure tank level is non-negative
    
    xp = (F_in - C * u * np.sqrt(x)) / A
    
    return xp